include/drivers/dma.h

```diff
@@ -327,6 +341,52 @@ static inline int z_impl_dma_stop(const struct device *dev, uint32_t channel)
 	return api->stop(dev, channel);
 }
 
+/**
+ * @brief request DMA channel.
+ *
+ * request DMA channel resources
+ * return -EINVAL if there is no valid channel available.
+ *
+ * @param dev     Pointer to the device structure for the driver instance.
+ * @param filter  channel filter
```

Comment:
What does channel filter mean here?

Reply:
in our driver channel 0-3 can act as periodic channel and others can not. so I add this filter.